iotc-azrtos-sdk/src/iotconnect_di.c

```diff
@@ -23,7 +21,9 @@
 
 #define URL_RESOUCE_BUFFER_SIZE 40
 
-#define WORK_BUFFER_SIZE (TO_CERT_X509_MAXSIZE * 2 + 1)
+// some certs can be more than 512 bytes, so we ought to have enough space for all of 
+// it the hex characters. 
+#define WORK_BUFFER_SIZE (1500) 
```

Comment:
Would it be better to extend TO_CERT_X509_MAXSIZE to an appropriate larger size (not sure if something else depends on it being a specific size though?), or create a new larger define